substrate/internal/defs

```diff
@@ -92,14 +92,8 @@
 
 #if __cplusplus >= 201402L
 #	define SUBSTRATE_CXX14_CONSTEXPR constexpr
-#	if defined (_WIN32)
```

Comment:
We recall having to have this so older MSVC would work correctly due to patchy `constexpr` support even when in C++14 mode - is that resolved now?

Reply:
Yes, as demonstrated in CI-- 2019 and 2022 all pass with 14. I can test tomorrow with 2017 if needed.

Reply:
For future reference, these were already no-ops because their use was removed in https://github.com/bad-alloc-heavy-industries/substrate/commit/c4e6194f570c9f59f617ed9110ab6b594c0478ed to no harm.